pkg/reconciler/v1alpha1/taskrun/taskrun.go

```diff
@@ -202,20 +204,25 @@ func (c *Reconciler) reconcile(ctx context.Context, tr *v1alpha1.TaskRun) error
 	// 	return nil
 	// }
 
-	// sync build status with taskrun status, if it exists
+	before := tr.Status.GetCondition(duckv1alpha1.ConditionSucceeded)
+	// sync build status with taskrun status
 	if build.Status.GetCondition(duckv1alpha1.ConditionSucceeded) != nil {
 		tr.Status.SetCondition(build.Status.GetCondition(duckv1alpha1.ConditionSucceeded))
 	} else {
 		// If the Build status doesn't exist yet, it's because we just started running
 		tr.Status.SetCondition(&duckv1alpha1.Condition{
-			Type:   duckv1alpha1.ConditionSucceeded,
-			Status: corev1.ConditionUnknown,
-			Reason: ReasonRunning,
+			Type:    duckv1alpha1.ConditionSucceeded,
+			Status:  corev1.ConditionUnknown,
+			Reason:  ReasonRunning,
+			Message: ReasonRunning,
 		})
 	}
 
+	after := tr.Status.GetCondition(duckv1alpha1.ConditionSucceeded)
+	c.Base.EmitEvent(before, after, tr)
```

Comment:
oh dear, it feels like we're sneaking inheritance in here

as a much longer story, i am definitely not a fan of inheritance, and afaik Go was designed in such a way to purposefully avoid inheritance

how about if the `EmitEvent` function took `b.Recorder` as a parameter instead?

Reply:
In that case I'd move it to another file also, like maybe an `events` package somewhere

Reply:
good comment!

Reply:
Sure I will create separate function(and add tests) but still under reconciler package because it holds relevance only in that package as of now. If there are more use cases outside then I would agree with you on separate package.